src/main/kotlin/org/jetbrains/research/testspark/services/ReportLockingService.kt

```diff
@@ -24,14 +25,15 @@ class ReportLockingService(private val project: Project) {
      * Updates the state after the action of publishing results.
      * @param report the generated test suite
      */
-    fun receiveReport(report: Report) {
-        this.report = report
+    fun receiveReport(result: UIContext?) {
+        this.report = result!!.testGenerationOutput.testGenerationResultList[0]!!
```

Comment:
When we started to implement this part for the first time, we envisioned the time that the user could ask multiple test generators to generate tests simultaneously, and then TestSpark would show generated tests by all of the techniques in a merged report. However, users can now ask for test generation using only one tool. This is why we use only the first element because there is only one report for one tool. In the future, when we add support for hybrid test generation, we need to review the report processing code as well